code/modules/mob/living/carbon/carbon_defense.dm

```diff
@@ -260,10 +260,12 @@
 	var/mob/living/carbon/target_collateral_carbon
 	var/obj/structure/table/target_table
 	var/obj/machinery/disposal/bin/target_disposal_bin
+	var/obj/structure/closet/target_storage_container
 	var/shove_blocked = FALSE //Used to check if a shove is blocked so that if it is knockdown logic can be applied
 
 	//Thank you based whoneedsspace
 	target_collateral_carbon = locate(/mob/living/carbon) in target_shove_turf.contents
+	target_storage_container = locate(/obj/structure/closet) in target_shove_turf.contents
```

Comment:
This is part of why this needs to be signal based.

We're performing *two* O(n) searches for *every* shove. The more we add, the slower this is going to get.